es-ds-components/components/es-video.vue

```diff
@@ -0,0 +1,73 @@
+<script setup lang="ts">
+import { useElementSize } from '@vueuse/core';
+
+interface IProps {
+    altText: string;
+    coverImageUrl: string;
+    embedUrl: string;
+}
+const props = defineProps<IProps>();
+const showVideo = ref(false);
+function getVideoIdFromUrl(url: string) {
```

Comment:
Nit: this is all kind of jammed together - thoughts on adding blank lines to separate semantic sections from each other? e.g. props/ref definitions from functions here, functions from const variables below, etc.